liquidator/liquidator.js

```diff
@@ -46,6 +46,27 @@ class Liquidator {
 
     let liquidationPromises = []; // store all promises to resolve in parallel later on.
     for (const position of underCollateralizedPositions) {
+      // Make sure that the bot has enough inventory/approval to do the liquidation.
+      const liquidation = this.empContract.methods.createLiquidation(
+        position.sponsor,
+        { rawValue: this.web3.utils.toWei(priceFeed) },
+        { rawValue: position.numTokens }
+      );
+
+      try {
+        await liquidation.call({ from: this.account });
+      } catch (error) {
```

Comment:
One thing we should consider is whether we should catch errors at the time of pushing them into `liquidationPromises`. We could do that by adding a `.on(error, () => {})` statement.

I think this would be useful because currently, the entire script will crash if a single transaction throws, now this PR prevents the script from crashing due to insufficient allowances/balances, but there are other low-level reasons why txns could revert like out of gas, front-running, nonce errors, etc.

Catching individual errors without crashing the script would have prevented this bot from crashing during network outages, such as during the recent MakerDAO situation. 

Additionally, catching errors at the time of pushing them into the promises array means that we don't need to think of and catch every single type of error.

Reply:
I agree that we should do some sort of error handling on the individual transactions. I have a TODO for that in the dispute bot. Will open an issue for that.

Reply:
However, I think catching errors this way is still very useful for a few reasons:
1. Because if there's a *logical* issue with the transaction, we don't waste ETH by trying to send it to the network.
2. There is some use in differentiating logic errors from transaction errors. We can aid the debugging process by differentiating the two cases.

Reply:
See issue #1114.